### PR TITLE
Setting IPAddressesEnabled to disable sending IP and machine name to BE. 

### PR DESCRIPTION
**Pull request by EmilianoSanchez**
- Setting IPAddressesEnabled only for Nodejs. True by default.
- Setting value used as condition for sending headers 'SplitSDKMachineIP' with IP address, and 'SplitSDKMachineName' with machine name.

### CODE REVIEW

File: types/splitio.d.ts [hunk 1]
```diff
@@ -38,7 +38,8 @@ interface ISettings {
     authorizationKey: string,
     key: SplitIO.SplitKey,
     trafficType: string,
-    labelsEnabled: boolean
+    labelsEnabled: boolean,
+    IPAddressesEnabled: boolean
```
Comment (NicoZelaya) [inline types/splitio.d.ts]: Let's add a test for it too.

File: src/__tests__/nodeSuites/ip-addresses-setting.spec.js [hunk 2]
```diff
@@ -52,44 +54,52 @@ const configWithIPAddressesDefault = {
...
-  });
+  function mockAndAssertIPAddressesEnabled(config, endpointToMock, clientAction){
+    finish.next();
 
-  clientDisabled.track('nicolas@split.io', 'sometraffictype', 'someEvent', 10);
+    const splitio = SplitFactory(config);
+    const client = splitio.client();
+    const settings = SettingsFactory(config);
+  
+    mock.onPost(settings.url(endpointToMock)).replyOnce(req => {
```
Comment (NicoZelaya) [inline src/__tests__/nodeSuites/ip-addresses-setting.spec.js]: We should try to mock all endpoints where we post data, for events, impressions and both for telemetry (counts and times).

You can either use a generator function or keep track of these with a counter of some sorts, as long as you do it inside the `mockAndAssertIPAddressesEnabled` function.

File: src/__tests__/nodeSuites/ip-addresses-setting.spec.js [hunk 3]
```diff
@@ -52,44 +54,52 @@ const configWithIPAddressesDefault = {
...
+    }while(count);
+    clients.forEach(client => client.destroy());
+    assert.end();
+  })();
 
-  let splitioDisabled = SplitFactory(configWithIPAddressesDisabled);
-  let clientDisabled = splitioDisabled.client();
-  const settingsDisabled = SettingsFactory(configWithIPAddressesDisabled);
+  function assertHeaders(IPAddressesEnabled, req){
+    assert.equal(HEADER_SPLITSDKMACHINEIP in req.headers, IPAddressesEnabled, `Request must include ${HEADER_SPLITSDKMACHINEIP} header if IPAddressesEnabled is true.`);
```
Comment (NicoZelaya) [inline src/__tests__/nodeSuites/ip-addresses-setting.spec.js]: Truly a minor thing but the assertion message is missleading (it always talks about the value being true). Maybe we can have something like:
`Request must ${IPAddressesEnabled ? '' : 'NOT '}include ${HEADER_SPLITSDKMACHINEIP} header if IPAddressesEnabled is ${IPAddressesEnabled}.`

**Comment by NicoZelaya** [comment]
Let's add a similar integration suite for Browsers to validate the IP is not sent regardless of what's on the config object.

File: src/__tests__/nodeSuites/ip-addresses-setting.spec.js (hunk 2 above)
Comment (EmilianoSanchez) [reply src/__tests__/nodeSuites/ip-addresses-setting.spec.js]: Done

File: src/__tests__/nodeSuites/ip-addresses-setting.spec.js (hunk 3 above)
Comment (EmilianoSanchez) [reply src/__tests__/nodeSuites/ip-addresses-setting.spec.js]: Done

**Comment by NicoZelaya** [comment]
LGTM, @EmilianoSanchez go ahead and merge whenever you want

**Comment by EmilianoSanchez** [comment]
* Fixed typescript definition (Removed 'LOCALSTORAGE' value from NodeSyncStorage type).
* Include unit and integration tests.

**Pull request merged**